### PR TITLE
Adds the bigip_selfip module

### PR DESCRIPTION
**Pull request by caphrim007**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

network/f5/bigip_selfip.py
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### SUMMARY

Another bootstrapping module, this module allows for one to manage
self IP addresses on a BIG-IP.

Tests for this module can be found here

https://github.com/F5Networks/f5-ansible/blob/master/roles/__bigip_selfip/tasks/main.yaml

Platforms this was tested on are

11.5.4 HF1
11.6.0
12.0.0
12.1.0 HF1

### CODE REVIEW

**Comment by caphrim007** [comment]
@gundalow

**Comment by gregdek** [comment]
Thanks again to @caphrim007. This module is going into community review.

We notice that you are one of the maintainers (@mhite, @caphrim007) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/f5/bigip_selfip.py [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+version_added: "2.2"
+options:
+  address:
+    description:
+      - The IP addresses for the new self IP. This value is ignored upon update
+        as addresses themselves cannot be changed after they are created.
+  name:
+    description:
+      - The self IP to create
+    required: false
```
Comment (gundalow) [inline network/f5/bigip_selfip.py]: `name` is required according to `meta_args`.

File: network/f5/bigip_selfip.py [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+  - Manage Self-IPs on a BIG-IP system
+version_added: "2.2"
+options:
+  address:
+    description:
+      - The IP addresses for the new self IP. This value is ignored upon update
+        as addresses themselves cannot be changed after they are created.
+  name:
+    description:
+      - The self IP to create
```
Comment (gundalow) [inline network/f5/bigip_selfip.py]: Missing fullstop, same applies for some other `descriptions`.

File: network/f5/bigip_selfip.py [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+try:
+    from f5.bigip import ManagementRoot
+    from icontrol.session import iControlUnexpectedHTTPError
+    HAS_F5SDK = True
+except ImportError:
+    HAS_F5SDK = False
+
+try:
+    from netaddr import *
+    HAS_NETADDR = True
```
Comment (gundalow) [inline network/f5/bigip_selfip.py:141]: Where is `HAS_NETADDR` used?

**Comment by gundalow** [comment]
needs_revision - Just a few minor docs things

Your tests in https://github.com/F5Networks/f5-ansible/blob/master/roles/__bigip_selfip/tasks/main.yaml look good

File: network/f5/bigip_selfip.py (hunk 3 above)
Comment (mhite) [inline network/f5/bigip_selfip.py:141]: Looks like the module makes heavy use of IPAddr from netaddr. Ideally this module should display an error at runtime if it doesn't find the module.

File: network/f5/bigip_selfip.py [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+
+try:
+    from f5.bigip import ManagementRoot
+    from icontrol.session import iControlUnexpectedHTTPError
+    HAS_F5SDK = True
+except ImportError:
+    HAS_F5SDK = False
+
+try:
+    from netaddr import *
```
Comment (mhite) [inline network/f5/bigip_selfip.py]: Pet peeve: blind import of \* into the namespace. I personally feel `from netaddr import IPAddr` is safer and makes the code easier to read/maintain.

Comment (caphrim007) [inline network/f5/bigip_selfip.py]: @mhite I think you meant IPNetwork and AddrFormatError?

**Comment by caphrim007** [comment]
@gundalow @mhite addressed your comments.

ready_for_review

**Comment by mhite** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @caphrim007 for this PR, and thanks @mhite, @caphrim007 for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**